File: Lab_9/9/FocusControl.py
class FocusControl:
    def __init__(self, standart_length = 15, standart_minlength = 1, standart_maxlength = 30 ):
        self.__length = standart_length
        self.__minlength = standart_minlength
        self.__maxlength = standart_maxlength
    def scrollUp (self):
        if self.__length==self.__maxlength :
            print("Error. Length is already max")
        else:
            self.__length = self.length+1
    def scrollDown (self):
        if self.length == self.__minlength:
            print("Error. Length is already min")
        else:
            self.__length=self.__length+1
    def getFocusDistance(self):
        return self.__length
    def getMinFocusDistance(self):
        return self.__minlength
    def getMaxFocusDistance(self):
        return self.__maxlength

File: Lab_2/src/ki304/Kisil/Lab2/FocusControl.java
package ki304.Kisil.Lab2;


/**
 * Class <code>FocusControl</code> implement focus adjustment scroller
 */
public class FocusControl
{
	private int length;
	private int maxlength;
	private int minlength;
	
	/**
	 * Constructor
	 */
	 public FocusControl()
	 {
		 length = 15;
		 minlength = 1;
		 maxlength = 30;
	 }
	 
	 /**
	  * @param <code>len</code> Focus distance
	  */
	 public FocusControl(int len)
	 {
		 length = len;
		 minlength = 1;
		 maxlength = 30;
	 }
	 
	 /**
	  * @param <code>len</code> Focus distance
	  * @param <code>max</code> Max Focus distance
	  */
	 public FocusControl(int len, int max)
	 {
		 length = len;
		 minlength = 1;
		 maxlength = max;
	 }
	 /**
	  * Method simulate increasing focus distance
	  */
	 public boolean scrollUp()
	 {
		 if(length == maxlength)
		 {
			 System.out.print("Error (min length)"+"\n");
		 return false;}
		 else 
		 {
		 length=length+1;
		 return true;
		 }
	 }
	 /**
	  * Method simulate decreasing focus distance
	  */
	 public boolean scrollDown()
	 {
		 if(length == minlength)
		 {
			 System.out.print("Error (min length)"+"\n");
		 return false;
		 }
		 else 
		 {
		 length=length-1;
		 return true;	
		 }

	 }
	 /**
	  * Method returns focus distance at the moment
	  * @returns current focus distance 
	  */
	 public int getFocusDistance()
	 {
		 return length;
	 }
	 /**
	  * Method returns max focus distance
	  * @returns current focus distance 
	  */
	 public int getMaxFocusDistance()
	 {
		 return maxlength;
	 }
	 /**
	  * Method returns min focus distance
	  * @returns current focus distance 
	  */
	 public int getMinFocusDistance()
	 {
		 return minlength;
	 }
}
